.pre-commit-config.yaml

```diff
@@ -38,10 +38,25 @@ repos:
                 notebooks/benchmarks/.*.ipynb|
                 notebooks/genomic_examples/.*.ipynb|
             )$
+
 - repo: https://github.com/astral-sh/ruff-pre-commit
   rev: v0.0.287
   hooks:
   - id: ruff
     name: ruff (fixed notebooks)
     types_or: [jupyter]
     files: *cleanNotebooks
+
+- repo: local
```

Comment:
I'm in two minds on whether it's worth implementing the watermark as a pre-commit hook for notebooks. Pre-commit hooks are certainly nice for linting, but this might be a bit to strict.

Watermarks make sense for fairly long examples, especially those that use heavy 3rd party libraries, take a while to execute, or might be hard to reproduce / be slow running. However, for short API demos or introductory pages I think the watermark doesn't add much value, and is not worth the added visual clutter. So, not sure it's wise to add it as an absolutely mandatory hook for all notebooks.

In any case, I would suggest removing or commenting this hook out for now, so we can merge the PR and get the contributing guidelines published ASAP.